libbeat/scripts/kibana_extract.py

```diff
@@ -0,0 +1,144 @@
+
+import os
+import json
+import jinja2
+
+
+def unescape_json(escaped):
+    obj = json.loads(escaped)
+    return json.dumps(obj, sort_keys=True, indent=4, separators=(',', ': '))
+
+def load_json(filepath):
+    json_data=open(filepath).read()
+    return json.loads(json_data)
+
+def load_search_source(file):
+    data = load_json(file)
+
+    field = "searchSourceJSON"
+
+    if data.has_key("kibanaSavedObjectMeta") and data["kibanaSavedObjectMeta"].has_key(field):
+        jsonData = unescape_json(data["kibanaSavedObjectMeta"][field])
+    else:
+        return ""
+
+    write_output(file, field, jsonData)
+
+
+def load_field(file, field):
+    data = load_json(file)
+
+    if  data.has_key(field):
+        jsonData = unescape_json(data[field])
+    else:
+        return ""
+
+    write_output(file, field, jsonData)
+
+
+def write_output(file, field, jsonData):
+
+    dir_name = os.path.splitext(file)[0]
+
+    if not os.path.exists(dir_name):
+        os.mkdir(dir_name)
+
+    with open(dir_name + "/" + field + ".json", 'w') as outfile:
+        outfile.write(jsonData)
+
+
+def extract():
+    base = './etc/kibana'
+
+    folders = os.listdir(base)
```

Comment:
* Lets try to use beatname-module for dashboards and just beatname if there is only 1
* Should be try beatname-visualization and change it in case we hit some conflicts?
* There is one shared visualization with the markdown I think, but not searches.